### PR TITLE
📖  update release process docs to match current steps

### LINKED ISSUE #475: Update release documentation
The release docs, https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/master/docs/releasing.md are out of date. We should update them to match the current release process.

### PR DESCRIPTION
**Pull request by nader-ziada**
**What this PR does / why we need it**:
- update release process docs
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #475 

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [x] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
Update document for release process  
```

/cc @CecileRobertMichon

### CODE REVIEW

File: docs/releasing.md [hunk 1]
```diff
@@ -1,69 +1,42 @@
...
-
-    REGISTRY="quay.io/k8s" \
-    MANAGER_IMAGE_TAG="<version>" \
-    time make docker-push
-    ```
-11. Publish release
-12. [Announce][release-announcement] the release
+## Change milestone 	
+ - Change milestone applier so new changes can be applied to the appropriate lease 
+      - Open a PR in https://github.com/kubernetes/test-infra to change the llowing line https://github.com/kubernetes/test-infra/blob/master/nfig/prow/plugins.yaml#L323
```
Comment (CecileRobertMichon) [inline docs/releasing.md:5]:
```suggestion
      - Open a PR in https://github.com/kubernetes/test-infra to change the following line https://github.com/kubernetes/test-infra/blob/master/nfig/prow/plugins.yaml#L323
```

File: docs/releasing.md [hunk 2]
```diff
@@ -1,69 +1,42 @@
...
-
-    REGISTRY="quay.io/k8s" \
-    MANAGER_IMAGE_TAG="<version>" \
-    time make docker-push
-    ```
-11. Publish release
-12. [Announce][release-announcement] the release
+## Change milestone 	
+ - Change milestone applier so new changes can be applied to the appropriate lease 
+      - Open a PR in https://github.com/kubernetes/test-infra to change the following line https://github.com/kubernetes/test-infra/blob/master/config/prow/plugins.yaml#L323
```
Comment (CecileRobertMichon) [inline docs/releasing.md:5]:
```suggestion
      - Open a PR in https://github.com/kubernetes/test-infra to change this [line](https://github.com/kubernetes/test-infra/blob/25db54eb9d52e08c16b3601726d8f154f8741025/config/prow/plugins.yaml#L344)
```

the line number might change in the master branch

File: docs/releasing.md [hunk 3]
```diff
@@ -1,69 +1,42 @@
...
+- Example PR: https://github.com/kubernetes/test-infra/pull/16827
+
+## Prepare branch, tag and release-notes	
+ - Identify a known good commit on the main branch
+ - Fast forward release branch to selected commit, always release from the is branch and not from master
```
Comment (CecileRobertMichon) [inline docs/releasing.md:10]:
```suggestion
 - Fast forward release branch to selected commit, always release from the release branch and not from master
```

File: docs/releasing.md [hunk 4]
```diff
@@ -1,69 +1,42 @@
...
+- git fetch upstream
+- git merge --ff-only upstream/master
+ - Create tag with git 
+- export RELEASE_TAG=v0.4.6 # the tag of the release to be cut
+- git tag -s ${RELEASE_TAG} -m "${RELEASE_TAG}"
```
Comment (CecileRobertMichon) [inline docs/releasing.md:15]: maybe we should add a note about setting up a GPG key with your github account as a pre-req to be able to sign the tag

https://docs.github.com/en/enterprise/2.16/user/github/authenticating-to-github/generating-a-new-gpg-key

Comment (CecileRobertMichon) [reply docs/releasing.md:15]: ah just read the line below, please ignore me :)

File: docs/releasing.md [hunk 5]
```diff
@@ -1,69 +1,42 @@
...
+	- -s creates a signed tag, user must have GPG added to heir github count
```
Comment (CecileRobertMichon) [inline docs/releasing.md:16]:
```suggestion
	- -s creates a signed tag, you must have a GPG key [added to your GitHub account](https://docs.github.com/en/enterprise/2.16/user/github/authenticating-to-github/generating-a-new-gpg-key)
```

File: docs/releasing.md [hunk 6]
```diff
@@ -1,69 +1,42 @@
...
+- git push upstream ${RELEASE_TAG}
+ `make release` from repo, this will create the release artifacts in the t/ folder 
+ Release notes https://github.com/kubernetes/release/blob/master/cmd/lease-notes/README.md
```
Comment (CecileRobertMichon) [inline docs/releasing.md:19]:
```suggestion
 Release notes https://github.com/kubernetes/release/blob/master/cmd/release-notes/README.md
```

File: docs/releasing.md [hunk 7]
```diff
@@ -1,69 +1,42 @@
...
+- Export GITHUB_TOKEN
+release-notes --github-org kubernetes-sigs --github-repo uster-api-provider-azure \
```
Comment (CecileRobertMichon) [inline docs/releasing.md:21]:
```suggestion
release-notes --github-org kubernetes-sigs --github-repo cluster-api-provider-azure \                
```

File: docs/releasing.md [hunk 8]
```diff
@@ -1,69 +1,42 @@
...
+ Release notes https://github.com/kubernetes/release/blob/master/cmd/lease-notes/README.md
+- Export GITHUB_TOKEN
+release-notes --github-org kubernetes-sigs --github-repo uster-api-provider-azure \                
+ --start-sha 1cf1ec4a1effd9340fe7370ab45b173a4979dc8f  \
+  --end-sha e843409f896981185ca31d6b4a4c939f27d975de 
+  (commits range from first commit after previous release to new release commit)
+- Manually format and categorize
+
+## Promote image to prod repo	
+ Promote image (how is the image built?)
```
Comment (CecileRobertMichon) [inline docs/releasing.md:28]: (how is the image built?) --> https://testgrid.k8s.io/sig-cluster-lifecycle-cluster-api-provider-azure#post-cluster-api-provider-azure-push-images

File: docs/releasing.md [hunk 9]
```diff
@@ -1,69 +1,42 @@
...
+ - Create a PR in https://github.com/kubernetes/k8s.io to update the image a and tag
```
Comment (CecileRobertMichon) [inline docs/releasing.md:29]:
```suggestion
 - Create a PR in https://github.com/kubernetes/k8s.io to add the image and tag
```

File: docs/releasing.md [hunk 10]
```diff
@@ -1,69 +1,42 @@
...
+ - Example PR: https://github.com/kubernetes/k8s.io/pull/1030/files
+ - Location of image: https://console.cloud.google.com/gcr/images/s-staging-cluster-api-azure/GLOBAL/cluster-api-azure-controller?rImageListsize=30
+
+## Release in github	
+ Create the github release in the UI
+ - Create a draft release in GitHub and associate it with the tag that was st created
```
Comment (CecileRobertMichon) [inline docs/releasing.md:35]:
```suggestion
 - Create a draft release in GitHub and associate it with the tag that was created
```

Comment (nader-ziada) [reply docs/releasing.md:35]: something went wrong with my formatting, there are a bunch of words cut off, will go through it all again

**Comment by nader-ziada** [comment]
@CecileRobertMichon updated based on comments

File: docs/releasing.md [hunk 11]
```diff
@@ -1,69 +1,45 @@
...
-   - An [OWNER](/OWNERS) runs `git tag -s <version>-rc.n`, inserts the changelog and pushes the tag with `git push <version>-rc.n`
-- Review and discuss any potential modifications to the [support matrix/policy][support-policy]
-- Review the development, getting started documentation, generation scripts, and update as necessary
-- Run `make release-artifacts`, with the appropriate environment variables:
+## Change milestone 	
+ - Change milestone applier so new changes can be applied to the appropriate release 
+      - Open a PR in https://github.com/kubernetes/test-infra to change this [line](https://github.com/kubernetes/test-infra/blob/25db54eb9d52e08c16b3601726d8f154f8741025/config/prow/plugins.yaml#L344)
+- Example PR: https://github.com/kubernetes/test-infra/pull/16827
+
+## Prepare branch, tag and release-notes
```
Comment (mboersma) [inline docs/releasing.md:8]:
```suggestion
## Prepare branch, tag and release notes
```

File: docs/releasing.md [hunk 12]
```diff
@@ -1,69 +1,45 @@
...
-- Review the development, getting started documentation, generation scripts, and update as necessary
-- Run `make release-artifacts`, with the appropriate environment variables:
+## Change milestone 	
+ - Change milestone applier so new changes can be applied to the appropriate release 
+      - Open a PR in https://github.com/kubernetes/test-infra to change this [line](https://github.com/kubernetes/test-infra/blob/25db54eb9d52e08c16b3601726d8f154f8741025/config/prow/plugins.yaml#L344)
+- Example PR: https://github.com/kubernetes/test-infra/pull/16827
+
+## Prepare branch, tag and release-notes	
+ - Identify a known good commit on the main branch
+ - Fast forward release branch to the selected commit, always release from the release branch and not from master
```
Comment (mboersma) [inline docs/releasing.md:10]:
```suggestion
 - Fast-forward the release branch to the selected commit. Always release from the release branch and not from master!
```

File: docs/releasing.md [hunk 13]
```diff
@@ -1,69 +1,45 @@
...
+   - git fetch upstream
```
Comment (mboersma) [inline docs/releasing.md:11]:
```suggestion
   - `git fetch upstream`
```

File: docs/releasing.md [hunk 14]
```diff
@@ -1,69 +1,45 @@
...
+   - git merge --ff-only upstream/master
```
Comment (mboersma) [inline docs/releasing.md:12]:
```suggestion
   - `git merge --ff-only upstream/master`
```

File: docs/releasing.md [hunk 15]
```diff
@@ -1,69 +1,45 @@
...
+ - Create tag with git 
+   - export RELEASE_TAG=v0.4.6 # the tag of the release to be cut
```
Comment (mboersma) [inline docs/releasing.md:14]:
```suggestion
   - `export RELEASE_TAG=v0.4.6` (the tag of the release to be cut)
```

File: docs/releasing.md [hunk 16]
```diff
@@ -1,69 +1,45 @@
...
+   - git tag -s ${RELEASE_TAG} -m "${RELEASE_TAG}"
```
Comment (mboersma) [inline docs/releasing.md:15]:
```suggestion
   - `git tag -s ${RELEASE_TAG} -m "${RELEASE_TAG}"`
```

File: docs/releasing.md [hunk 17]
```diff
@@ -1,69 +1,45 @@
...
+	 - -s creates a signed tag, you must have a GPG key [added to your GitHub account](https://docs.github.com/en/enterprise/2.16/user/github/authenticating-to-github/generating-a-new-gpg-key)
+   - git push upstream ${RELEASE_TAG}
```
Comment (mboersma) [inline docs/releasing.md:17]:
```suggestion
   - `git push upstream ${RELEASE_TAG}`
```

File: docs/releasing.md [hunk 18]
```diff
@@ -1,69 +1,45 @@
...
+ - `make release` from repo, this will create the release artifacts in the `out/` folder 
+ - Release notes https://github.com/kubernetes/release/blob/master/cmd/release-notes/README.md
+ - Export GITHUB_TOKEN
+ - Run the release-notes tool with the appropriate commits # commits range from first commit after the previous release to new release commit
```
Comment (mboersma) [inline docs/releasing.md:21]:
```suggestion
 - Run the `release-notes` tool with the appropriate commits. Commits range from the first commit after the previous release to the new release commit. 
```

File: docs/releasing.md [hunk 19]
```diff
@@ -1,69 +1,45 @@
...
+ - Fast forward release branch to the selected commit, always release from the release branch and not from master
+   - git fetch upstream
+   - git merge --ff-only upstream/master
+ - Create tag with git 
+   - export RELEASE_TAG=v0.4.6 # the tag of the release to be cut
+   - git tag -s ${RELEASE_TAG} -m "${RELEASE_TAG}"
+	 - -s creates a signed tag, you must have a GPG key [added to your GitHub account](https://docs.github.com/en/enterprise/2.16/user/github/authenticating-to-github/generating-a-new-gpg-key)
+   - git push upstream ${RELEASE_TAG}
+ - `make release` from repo, this will create the release artifacts in the `out/` folder 
+ - Release notes https://github.com/kubernetes/release/blob/master/cmd/release-notes/README.md
```
Comment (mboersma) [inline docs/releasing.md:19]:
```suggestion
 - To install the `release-notes` tool, see https://github.com/kubernetes/release/blob/master/cmd/release-notes/README.md
```

File: docs/releasing.md [hunk 20]
```diff
@@ -1,69 +1,45 @@
...
-    MANAGER_IMAGE_TAG="<version>" \
-    time make docker-push
-    ```
-11. Publish release
-12. [Announce][release-announcement] the release
+ - Manually format and categorize the release notes
+
+## Promote image to prod repo	
+ Promote image
+ - Images are build by the [post push images job](https://testgrid.k8s.io/sig-cluster-lifecycle-cluster-api-provider-azure#post-cluster-api-provider-azure-push-images)
```
Comment (mboersma) [inline docs/releasing.md:31]:
```suggestion
 - Images are built by the [post push images job](https://testgrid.k8s.io/sig-cluster-lifecycle-cluster-api-provider-azure#post-cluster-api-provider-azure-push-images)
```

File: docs/releasing.md [hunk 21]
```diff
@@ -1,69 +1,45 @@
...
+ - Create a PR in https://github.com/kubernetes/k8s.io to add the image and tag
+   - Example PR: https://github.com/kubernetes/k8s.io/pull/1030/files
+ - Location of image: https://console.cloud.google.com/gcr/images/s-staging-cluster-api-azure/GLOBAL/cluster-api-azure-controller?rImageListsize=30
+
+## Release in github
```
Comment (mboersma) [inline docs/releasing.md:36]:
```suggestion
## Release in GitHub
```

File: docs/releasing.md [hunk 22]
```diff
@@ -1,69 +1,45 @@
...
+
+## Promote image to prod repo	
+ Promote image
+ - Images are build by the [post push images job](https://testgrid.k8s.io/sig-cluster-lifecycle-cluster-api-provider-azure#post-cluster-api-provider-azure-push-images)
+ - Create a PR in https://github.com/kubernetes/k8s.io to add the image and tag
+   - Example PR: https://github.com/kubernetes/k8s.io/pull/1030/files
+ - Location of image: https://console.cloud.google.com/gcr/images/s-staging-cluster-api-azure/GLOBAL/cluster-api-azure-controller?rImageListsize=30
+
+## Release in github	
+ Create the github release in the UI
```
Comment (mboersma) [inline docs/releasing.md:37]:
```suggestion
 Create the GitHub release in the UI
```

**Comment by nader-ziada** [comment]
I don't think I broke the e2e test in this PR :)

**Pull request merged**